bundle/config/mutator/resolve_resource_references.go

```diff
@@ -0,0 +1,48 @@
+package mutator
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/log"
+	"golang.org/x/sync/errgroup"
+)
+
+type resolveResourceReferences struct{}
+
+func ResolveResourceReferences() bundle.Mutator {
+	return &resolveResourceReferences{}
+}
+
+func (m *resolveResourceReferences) Apply(ctx context.Context, b *bundle.Bundle) error {
+	errs, errCtx := errgroup.WithContext(ctx)
+
+	for k := range b.Config.Variables {
+		v := b.Config.Variables[k]
```

Comment:
v can be nil as well (can happen if the variable is defined but doesn't have a default or a description)